### PR TITLE
feat(pdf): Update pdf.js to v2.2.228 and closure compiler to v20190729

### PR DESCRIPTION
**Pull request by jstoffan**
**Summary**
This changeset is partially based on and supersedes #810 and #908. I've migrated our calls to pdf.js to match the associated API changes in the latest version. However, there is no official migration guide, so further manual testing on document variations will be required.

**Known Issues**
- [ ] **Trivial**: Resizing the browser causes a harmless `TextLayer task cancelled` console error 
- [x] ~~**Blocker**: Find controls do not work properly and do not highlight search results~~

**Features Tested**
- Annotations
- Find controls and keyboard shortcuts
- Documents with digital signatures
- Documents with single and multiple pages
- Presentations with single and multiple pages
- Fullscreen mode via on-screen controls
- Page-up/down via on-screen controls and scroll wheel
- Text layer highlighting via click-drag and double-click
- Zoom-in/out via on-screen controls
- Thumbnail sidebar

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

File: src/lib/util.js [hunk 1]
```diff
@@ -5,14 +5,25 @@ import Location from './Location';
...
+const HEADER_CLIENT_VERSION = 'X-Box-Client-Version';
+const PROMISE_MAP = {};
+
+/**
+ * Clears the promise map of any active promises
+ *
+ * @private
+ * @return {void}
+ */
+export function clearPromises() {
```
Comment (jstoffan) [inline src/lib/util.js]: I'll add some tests for this new logic shortly.

File: src/lib/viewers/doc/__tests__/DocFindBar-test.js [hunk 2]
```diff
@@ -29,28 +28,20 @@ describe('lib/viewers/doc/DocFindBar', () => {
     beforeEach(() => {
         fixture.load('viewers/doc/__tests__/DocFindBar-test.html');
 
-        docEl = document.querySelector('.test-container');
+        eventBus = { off: sandbox.stub, on: sandbox.stub() };
```
Comment (mickr) [inline src/lib/viewers/doc/__tests__/DocFindBar-test.js]: Should `sandbox.stub` be called?

File: src/lib/viewers/doc/SinglePageViewer.js [hunk 3]
```diff
@@ -10,15 +10,10 @@ class SinglePageViewer extends DocumentViewer {
      *
      * @protected
      * @override
-     * @return {PDFJS.PDFViewer} PDF viewer type
+     * @return {Object} PDF viewer type
```
Comment (mickr) [inline src/lib/viewers/doc/SinglePageViewer.js]: Can you use the same view type for this?

**Pull request merged**